enterprise/frontend/src/embedding-sdk/README.md

```diff
@@ -1189,6 +1189,169 @@ async function fetchRequestToken(url) {
 const config = {fetchRequestToken};
 ```
 
+### Using with Next.js
+
+#### Using App Router
+
+Create a component that imports the `MetabaseProvider` and mark it a React Client component with "use client";
+
+```typescript jsx
+"use client";
+
+import { MetabaseProvider, StaticQuestion } from "@metabase/embedding-sdk-react";
+
+const config = {...}; // Your Metabase SDK configuration
+
+export default function MetabaseComponents() {
+  return (
+    <MetabaseProvider config={config}>
+      <StaticQuestion questionId={QUESTION_ID} />
+    </MetabaseProvider>
+  );
+```
+
+Make sure to use default export, as named export is not supported with this setup.
```

Comment:
Is this us or nextjs?

Reply:
Or it's for the import on line 1220?
I think it's technically possible: https://nextjs.org/docs/pages/building-your-application/optimizing/lazy-loading#with-named-exports
I think people will try named exports and it could fail, WDYT about anticipating their "want" and showcase that scenario directly?

Reply:
This is certainly weird, what is the error you got when trying to use named exports for next/dynamic?

Reply:
This is what costed me half a day. I couldn't find anything relate to this problem at all.

Reply:
> This is certainly weird, what is the error you got when trying to use named exports for next/dynamic?

When using named export, Next will complain that `document` is undefined as if the component is still rendered on the server for some reason.

Reply:
> Is this us or nextjs?

Kind of sure it's Next.js problem since this is about importing a library that uses browser API like `document` or `window`.

Reply:
I think the fact that we have one big bundle, somewhere in the bundle we probably have many references to window/document which are note inside a function so it probably crashed while trying to load those "modules"

Reply:
I've included the repo in the [PR description](https://github.com/metabase/metabase/pull/48569#issue-2579236787) if you guys want to see the error yourself.

> To test Next.js please use my forked repos:
> 
> * App Router: https://github.com/WiNloSt/next-app-template
> * Pages Router: https://github.com/WiNloSt/next-pages-template

Reply:
> I think the fact that we have one big bundle, somewhere in the bundle we probably have many references to window/document which are note inside a function so it probably crashed while trying to load those "modules"

That doesn't quite explain why the `default` export doesn't crash though. In my naive mind, default export is also a named export with the name `default`. And I've also tested
```ts
import("mymodule").then(module => module.default)
```

That it didn't work either 😭